src/test/java/lambda/part1/exercise/Exercise3.java

```diff
@@ -0,0 +1,65 @@
+package lambda.part1.exercise;
+
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise3 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + expression-lambda
+        Arrays.sort(persons,(o1,o2)->Integer.compare(o1.getAge(),o2.getAge()));
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + statement-lambda
+
+        Arrays.sort(persons,(o1,o2)->{return o1.getLastName().compareTo(o2.getLastName())==0
+            ?o1.getFirstName().compareTo(o2.getFirstName())
+            :o1.getLastName().compareTo(o2.getLastName());});
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicateLambda() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+        Person person = FluentIterable.from(persons).firstMatch(prs->prs.getAge()==30).get();
```

Comment:
Better use Ctrl+Alt+L (autoformat code) before commit & push. You can also select code and press Ctrl+Shift+Alt+L.

Reply:
I did it. But it did not give the effect. The only think i can try to make you happy is to press Enter before "firstMatch" method)))

Reply:
Thank you for the comments btw.